ert_shared/ensemble_evaluator/dispatch.py

```diff
@@ -3,28 +3,39 @@
 import time
 
 from collections import defaultdict, OrderedDict
-from typing import Optional
 
 from ert.ensemble_evaluator import identifiers
 
-from ert.ensemble_evaluator.state import (
-    ENSEMBLE_STATE_CANCELLED,
-    ENSEMBLE_STATE_FAILED,
-    ENSEMBLE_STATE_STOPPED,
-)
-
 logger = logging.getLogger(__name__)
 
 
-class Batcher:
-    def __init__(self, timeout, max_batch=1000, loop=None):
+class BatchingDispatcher:
+    def __init__(self, loop, timeout, max_batch=1000):
         self._timeout = timeout
         self._max_batch = max_batch
+
+        self._LOOKUP_MAP = defaultdict(list)
         self._running = True
         self._buffer = []
 
         # Schedule task
         self._task = asyncio.ensure_future(self._job(), loop=loop)
+        self._task.add_done_callback(self._done_callback)
+
+    def _done_callback(self, *args):
+        try:
+            failure = self._task.exception()
+            if failure is not None:
+                logger.warning(f"exception in batcher: {failure}")
+            else:
+                logger.debug("batcher finished normally")
+                return
+        except asyncio.CancelledError as ex:
+            failure = ex
```

Comment:
Unnecessary. Leftover from when you used `failure` in the outmost scope?

Reply:
Not really. `asyncio.Task.exception()` raises `CancellerError` if the task has been cancelled (i.e. `set_exception()` was not called). AFAICS this code covers all possibilities in a done-handler?

Reply:
I was referring to `failure = ex`

Reply:
Ahh... yes. Thx!